tests/CrawlerProcess/asyncio_deferred_signal.py

```diff
@@ -0,0 +1,45 @@
+import asyncio
+import sys
+
+import scrapy
+
+from scrapy.crawler import CrawlerProcess
+from twisted.internet.defer import Deferred
+
+
+class UppercasePipeline:
+    async def _open_spider(self, spider):
+        spider.logger.info("async pipeline opened!")
+        await asyncio.sleep(0.1)
+
+    def open_spider(self, spider):
+        loop = asyncio.get_event_loop()
+        return Deferred.fromFuture(loop.create_task(self._open_spider(spider)))
```

Comment:
What do you think about using scrapy.utils.defer.deferred_from_coro or scrapy.utils.defer.deferred_f_from_coro_f here, as @wRAR suggested?

Reply:
Done, thanks 🚀